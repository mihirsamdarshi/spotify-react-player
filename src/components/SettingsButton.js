import React from 'react';
import Button from '@material-ui/core/Button';
import Fab from '@material-ui/core/Fab';
import SettingsIcon from '@material-ui/icons/Settings';
import Dialog from '@material-ui/core/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import DialogActions from '@material-ui/core/DialogActions';

import '../stylesheets/SettingsButton.scss';

export default function SettingsButton() {
    const [state, setState] = React.useState({
        open: false,
        playbackService: 'Spotify',
        playlistService: 'Apple Music',
    });

    const handleClickOpen = () => {
        setState({
            ...state,
            open: true
        });
    };

    const handleClose = () => {
        setState({
            ...state,
            open: false
        });
    };

    const handleSwitch = () => {
        const [newPLS, newPBS] = [state.playbackService, state.playlistService];
        setState({
            ...state,
            playbackService: newPBS,
            playlistService: newPLS
        });
    };

    return (
        <div>
            <Fab size="small" className="settingsButton" onClick={handleClickOpen}>
                <SettingsIcon/>
            </Fab>
            <Dialog open={state.open} onClose={handleClose} className="classes.modal">
                <DialogTitle>Change Current Service</DialogTitle>
                <DialogContent>
                    <div className="serviceDialogContainer">
                        <span>
                            Music Playlists Come From:
                            {' '}
                            {state.playlistService}
                        </span>
                        <span>
                            Music Plays From:
                            {' '}
                            {state.playbackService}
                        </span>
                    </div>
                    <Button className="handleSwitchButton" onClick={handleSwitch} color="primary">
                        Switch Services
                    </Button>
                </DialogContent>
                <DialogActions>
                    <Button onClick={handleClose} color="primary">
                        Exit
                    </Button>
                </DialogActions>
            </Dialog>
        </div>
    );
}
