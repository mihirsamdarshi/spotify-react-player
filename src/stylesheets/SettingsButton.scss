.modal {
  width: 200px;
}

.handleSwitchButton {
  margin: auto;
  min-width: 120px;
}

.serviceDialogContainer {
  display: flex;
  flex-direction: column;
  column-gap: 5px;
  margin: auto;
  width: 300px;
  padding-bottom: 15px;
}