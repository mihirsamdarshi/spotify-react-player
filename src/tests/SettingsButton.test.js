import ReactDOM from 'react-dom';
import SettingsButton from '../components/SettingsButton';
import React from 'react';
import { shallow } from 'enzyme';

it('renders the Settings Button component', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SettingsButton />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('should match the snapshot', () => {
        const component = shallow(<SettingsButton />);
        expect(component).toMatchSnapshot();
});
