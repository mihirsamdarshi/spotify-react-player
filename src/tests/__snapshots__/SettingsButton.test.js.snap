// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Fab))
    className="settingsButton"
    onClick={[Function]}
    size="small"
  >
    <SettingsIcon />
  </WithStyles(ForwardRef(Fab))>
  <WithStyles(ForwardRef(Dialog))
    className="classes.modal"
    onClose={[Function]}
    open={false}
  >
    <WithStyles(ForwardRef(DialogTitle))>
      Change Current Service
    </WithStyles(ForwardRef(DialogTitle))>
    <WithStyles(ForwardRef(DialogContent))>
      <div
        className="serviceDialogContainer"
      >
        <span>
          Music Playlists Come From:
           
          Apple Music
        </span>
        <span>
          Music Plays From:
           
          Spotify
        </span>
      </div>
      <WithStyles(ForwardRef(Button))
        className="handleSwitchButton"
        color="primary"
        onClick={[Function]}
      >
        Switch Services
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogActions))>
      <WithStyles(ForwardRef(Button))
        color="primary"
        onClick={[Function]}
      >
        Exit
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(DialogActions))>
  </WithStyles(ForwardRef(Dialog))>
</div>
`;
